app/controllers/api/v1/study_files_controller.rb

```diff
@@ -354,30 +354,30 @@ def perform_update(study_file)
         study_file.delay.invalidate_cache_by_file_type
 
         # if a gene list or cluster got updated, we need to update the associated records
-        if safe_file_params[:file_type] == 'Gene List' && name_changed
-          precomputed_entry = PrecomputedScore.find_by(study_file_id: safe_file_params[:_id])
+        if study_file.file_type == 'Gene List' && name_changed
+          precomputed_entry = PrecomputedScore.find_by(study_file: study_file)
           if precomputed_entry.present?
-            logger.info "Updating gene list #{@precomputed_entry.name} to match #{safe_file_params[:name]}"
+            logger.info "Updating gene list #{precomputed_entry.name} to match #{study_file[:name]}"
             precomputed_entry.update(name: study_file.name)
           end
-        elsif safe_file_params[:file_type] == 'Cluster' && name_changed
-          cluster = ClusterGroup.find_by(study_file_id: safe_file_params[:_id])
+        elsif study_file.file_type == 'Cluster' && name_changed
+          cluster = ClusterGroup.find_by(study_file: study_file)
           if cluster.present?
-            logger.info "Updating cluster #{@cluster.name} to match #{safe_file_params[:name]}"
+            logger.info "Updating cluster #{cluster.name} to match #{study_file.name}"
             # before updating, check if the defaults also need to change
-            if @study.default_cluster == @cluster
-              @study.default_options[:cluster] = study_file.name
-              @study.save
+            if study.default_cluster == cluster
+              study.default_options[:cluster] = study_file.name
+              study.save
             end
-            @cluster.update(name: study_file.name)
+            cluster.update(name: study_file.name)
           end
         end
 
-        if ['Expression Matrix', 'MM Coordinate Matrix'].include?(safe_file_params[:file_type]) && !safe_file_params[:y_axis_label].blank?
+        if ['Expression Matrix', 'MMc Coordinate Matrix'].include?(study_file.file_type) && !safe_file_params[:y_axis_label].blank?
```

Comment:
Blocker: fix typo bug:

```suggestion
        if ['Expression Matrix', 'MM Coordinate Matrix'].include?(study_file.file_type) && !safe_file_params[:y_axis_label].blank?
```

(This isn't present in what was released yesterday, fortunately.)